c++/src/kj/async-io.c++

```diff
@@ -1123,6 +1123,7 @@ private:
     }
 
     ~BlockedPumpTo() noexcept(false) {
+      canceler.cancel(KJ_EXCEPTION(DISCONNECTED, "BlockedPumpTo was destroyed"));
```

Comment:
There's a lot of other cancelers in use in this file that present similar problems.

Maybe we should just make DISCONNECTED the default exception type for `Canceler` in general?